### PR TITLE
lint: Bump flake8 to 3.6.0

### PR DESCRIPTION
**Pull request by Empact**
Fix new lints in:
W504 line break after binary operator
W605 invalid escape sequence
F841 local variable 'e' is assigned to but never used

https://github.com/PyCQA/flake8/blob/master/docs/source/release-notes/3.6.0.rst

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Thanks for doing this!

**Comment by promag** [comment]
LGTM.

File: contrib/devtools/copyright_header.py [hunk 1]
```diff
@@ -87,20 +86,20 @@ def compile_copyright_regex(copyright_style, year_style, name):
     "Bitcoin Core Developers\n",
     "the Bitcoin Core developers\n",
     "The Bitcoin developers\n",
-    "The LevelDB Authors\. All rights reserved\.\n",
-    "BitPay Inc\.\n",
-    "BitPay, Inc\.\n",
-    "University of Illinois at Urbana-Champaign\.\n",
+    r"The LevelDB Authors\. All rights reserved\.\n",
```
Comment (laanwj) [inline contrib/devtools/copyright_header.py]: I don't think simply adding `r` here is correct.
```python
>>> "The LevelDB Authors\. All rights reserved\.\n" == r"The LevelDB Authors\. All rights reserved\.\n"
False
```
A raw string will contain *literal* \.\n`` which is not the intent here.

Comment (laanwj) [reply contrib/devtools/copyright_header.py]: the problem here is that the `.` should not be escaped, not that the string is not raw !

**Comment by Empact** [comment]
Planning to re-open later.

**Pull request closed without merging**